nemo/collections/asr/parts/k2/topologies.py

```diff
@@ -0,0 +1,136 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List
+
+import k2
+
+
+def build_topo(name: str, tokens: List[int], with_selfloops: bool = True) -> k2.Fsa:
+    """Helper function to build a topology.
+    Args:
+      name:
+        The topology name. Choices: default, compact, shared_blank, minimal
+      tokens:
+        A list of tokens, e.g., phones, characters, etc.
+      with_selfloops:
+        Whether to add token-to-epsilon self-loops to a topology
+    Returns:
+      Returns a topology FST.
+    """
+    if name == "default":
+        return build_default_topo(tokens, with_selfloops)
+    elif name == "compact":
+        return build_compact_topo(tokens, with_selfloops)
+    elif name == "shared_blank":
+        return build_shared_blank_topo(tokens, with_selfloops)
+    elif name == "minimal":
+        return build_minimal_topo(tokens)
+    else:
+        raise ValueError(f"Unknown topo name: {name}")
+
+
+def build_default_topo(tokens: List[int], with_selfloops: bool = True) -> k2.Fsa:
```

Comment:
Same as above

Reply:
Done in [7b2b4a4](https://github.com/NVIDIA/NeMo/pull/3351/commits/7b2b4a4ba4e34848d526d8c4ac53903bfbfe2fd2)